scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -292,8 +292,8 @@ object TreeSyntax {
                     case _ => false
                   } =>
                   true
-                // Include parentheses for `a op (bar: _*)`
-                case repeated: Term.Repeated =>
+                // Include parentheses for `a op (b: _*)`, tuple `a op ((b, c))`, `a op (_.bar), a op (_.opTwo(b))`, `a op (_ opTwo b)`
+                case _: Term.Repeated | _: Term.Tuple |  Term.Select(_: Term.Placeholder, _) | Term.Apply(Term.Select(_: Term.Placeholder, _), _) | Term.ApplyInfix(_: Term.Placeholder, _, _, _) =>
```

Comment:
Would it make sense to flip the logic around and not introduce parentheses for a whitelist of trees? Maybe it's enough to only skip parentheses for `_: Term.Ref`

Reply:
Hmm, good idea. I'll try it

Reply:
Ok, so I've (mostly) flipped it around. It's a little cleaner and should err on the side of adding parentheses in most cases. However, there's still some weird behaviour:

`list map (_ -> foo)` -> ` list map (_ -> foo)`
`list map (_ op foo)` -> ` list map ((_ op foo))` (Note the double parentheses)

I have no idea where this comes from, as far as I can tell [the AST's are the same](https://astexplorer.net/#/gist/38a1965132e6d507f974bd345c71caf1/df1fc34cbaa19b2c87e2087fd2856af24f141c62).

Reply:
Those parentheses could be caused by the pretty-printer for `Term.ApplyInfix`

Reply:
Isn't `_ -> foo` also a `Term.ApplyInfix` though? It seems to work there. With some testing it looks to me like the difference comes from using symbols or not

Reply:
It depends on how the pretty-printer interprets those trees, they can be printed differently even if the AST is identical. The code quality is quite bad in the pretty-printer, it was written before my time and I haven't touched it much.